bin/delete_lambda.py

```diff
@@ -10,9 +10,20 @@
 output = completed_process.stdout
 convert = json.loads(output.decode("utf-8").replace("'","'"))
 
-# Deletes ORCA lambda functions with the given prefix
 for sub in convert:
-    print("Deleting " + sub['FunctionName'])
-    lambda_output = sub['FunctionName']
-    delete_function = f"aws lambda delete-function --function-name {lambda_output}"
-    subprocess.run(delete_function, shell=True, capture_output=True)
+    lambda_output = sub['FunctionArn']
+    get_tags = f"aws lambda list-tags --resource {lambda_output}"
+    tags_process = subprocess.run(get_tags, shell=True, capture_output=True)
+    tag_output = tags_process.stdout
+    tag_convert = json.loads(tag_output.decode("utf-8").replace("'","'"))
+    try:
+        # Finds lambdas with the tag application set to ORCA and deletes the ORCA Lambdas
+        for tag in tag_convert:
+            if tag_convert[tag]['application'] == 'ORCA':
+                print("Deleting " + sub['FunctionName'])
+                del_lambda = sub['FunctionName']
+                delete_function = f"aws lambda delete-function --function-name {del_lambda}"
+                subprocess.run(delete_function, shell=True, capture_output=True)
+    # If the lambda with the provided prefix does not match the tag for deletion it is skipped and is not deleted
+    except KeyError:
+        pass
```

Comment:
If for some reason the delete API does not work, it will ignore the error. Recommend changing `pass` to raising an exception.

Reply:
Do you have the default profile set up for the cumulus account, I was able to find the lambdas matching that prefix and tag (took out the deletion when I ran the script just to list them and not delete). Alternatively I could add an input for a profile to use incase users have the default profile set for another account and can specify a cumulus profile
```
zroehrich@IGSKMN011LM0069 ~/Desktop/s3/test1 -  $ python3 delete.py
Enter Prefix: rhassanorca1
Deleting rhassanorca1_db_deploy
Deleting rhassanorca1_extract_filepaths_for_granule
Deleting rhassanorca1_post_to_database
Deleting rhassanorca1_post_copy_request_to_queue
Deleting rhassanorca1_internal_reconcile_report_orphan
Deleting rhassanorca1_delete_old_reconcile_jobs
Deleting rhassanorca1_request_from_archive
Deleting rhassanorca1_internal_reconcile_report_phantom
Deleting rhassanorca1_internal_reconcile_report_job
Deleting rhassanorca1_copy_from_archive
Deleting rhassanorca1_orca_catalog_reporting
Deleting rhassanorca1_copy_to_orca
Deleting rhassanorca1_request_status_for_job
Deleting rhassanorca1_internal_reconcile_report_mismatch
Deleting rhassanorca1_get_current_archive_list
Deleting rhassanorca1_perform_orca_reconcile
Deleting rhassanorca1_post_to_queue_and_trigger_step_function
Deleting rhassanorca1_request_status_for_granule
Deleting rhassanorca1_post_to_catalog
zroehrich@IGSKMN011LM0069 ~/Desktop/s3/test1 -  $ 
```

Reply:
My bad! I was connected to the DR account instead of the cumulus account which is why it could not find anything to delete with that prefix.

Looks good now.

Reply:
Okay I added in an input for an AWS profile to use so if users want to use a profile outside of default they can do that.
```
zroehrich@IGSKMN011LM0069 ~/Desktop/s3/test1 -  $ python3 delete.py
Enter Prefix: zr4
Enter AWS Profile: cumulus-sandbox
Deleting zr4_internal_reconcile_report_orphan
Deleting zr4_post_to_database
Deleting zr4_post_to_catalog
Deleting zr4_orca_catalog_reporting
Deleting zr4_request_status_for_job
Deleting zr4_perform_orca_reconcile
Deleting zr4_delete_old_reconcile_jobs
Deleting zr4_post_to_queue_and_trigger_step_function
Deleting zr4_get_current_archive_list
Deleting zr4_extract_filepaths_for_granule
Deleting zr4_internal_reconcile_report_job
Deleting zr4_copy_to_orca
Deleting zr4_db_deploy
Deleting zr4_post_copy_request_to_queue
Deleting zr4_request_from_archive
Deleting zr4_internal_reconcile_report_phantom
Deleting zr4_copy_from_archive
Deleting zr4_request_status_for_granule
Deleting zr4_internal_reconcile_report_mismatch
zroehrich@IGSKMN011LM0069 ~/Desktop/s3/test1 -  $ 
```